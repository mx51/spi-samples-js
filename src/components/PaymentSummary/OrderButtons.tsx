import { Button, Grid } from '@material-ui/core';
import React from 'react';
import { Link as LinkRouter } from 'react-router-dom';
import { useTransactionDetailPageStyle } from '../TransactionPage/TransactionDetailsPage/TransactionDetailsPage.style';
import {
  PATH_ACCOUNT_VERIFY,
  PATH_PRE_AUTH,
  PATH_PURCHASE,
  TEXT_PRE_AUTH,
} from '../../definitions/constants/routerConfigs';

type Props = {
  splitTransaction?: {
    currentSplitNumber: number;
    totalSplitNumber: number;
    amount: number;
    outstandingAmount: number;
    onClickNext: () => void;
  };
  typePath?: string;
  classes?: Record<string, string>;
};

const pathNameFormatter = (path: string) => {
  if (path) {
    const pathName = path.split('/')[1];
    return pathName.charAt(0).toUpperCase() + pathName.slice(1);
  }
  return '';
};

const SplitTransactionButton = ({ splitTransaction, classes }: Props) =>
  splitTransaction ? (
    <Button
      variant="contained"
      color="primary"
      size="large"
      classes={{ root: classes?.actionBtn }}
      onClick={splitTransaction.onClickNext}
    >
      {splitTransaction.currentSplitNumber >= splitTransaction.totalSplitNumber - 1
        ? 'Finish'
        : `Next (Split #${splitTransaction.currentSplitNumber + 2})`}
    </Button>
  ) : null;

const OtherTransactionButtons = ({ typePath, classes }: Props) => (
  <>
    {typePath && typePath !== PATH_PURCHASE ? (
      <Grid item xs={6}>
        <Button
          variant="contained"
          color="primary"
          size="large"
          classes={{ root: classes?.actionBtn }}
          component={LinkRouter}
          to={typePath !== PATH_ACCOUNT_VERIFY ? typePath : PATH_PRE_AUTH}
        >
          {typePath !== PATH_ACCOUNT_VERIFY ? pathNameFormatter(typePath) : TEXT_PRE_AUTH}
        </Button>
      </Grid>
    ) : null}
    <Grid item xs={typePath !== PATH_PURCHASE ? 6 : 12}>
      <Button
        data-test-id="newOrderBtn"
        variant="contained"
        color="primary"
        size="large"
        classes={{ root: classes?.actionBtn }}
        component={LinkRouter}
        to={PATH_PURCHASE}
      >
        New Order
      </Button>
    </Grid>
  </>
);

export const OrderButtons = ({ typePath, splitTransaction }: Props) => {
  const classes = useTransactionDetailPageStyle();

  return (
    <Grid container spacing={1}>
      {splitTransaction ? (
        <SplitTransactionButton splitTransaction={splitTransaction} classes={classes} />
      ) : (
        <OtherTransactionButtons typePath={typePath} classes={classes} />
      )}
    </Grid>
  );
};
