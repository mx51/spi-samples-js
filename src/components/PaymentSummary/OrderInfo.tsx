import { Box, Typography } from '@material-ui/core';
import React from 'react';
import { TxLogItem } from '../../services/txLogService';
import { useTransactionDetailPageStyle } from '../TransactionPage/TransactionDetailsPage/TransactionDetailsPage.style';

type Props = {
  currentTransaction: TxLogItem;
};

export const OrderInfo = ({ currentTransaction }: Props) => {
  const classes = useTransactionDetailPageStyle();
  const { posRefId, posId, tid, mid } = currentTransaction;

  return (
    <div className={classes.orderInfo}>
      <Typography component="code">
        <div>POS Ref ID:</div>
      </Typography>
      <Box bgcolor="rgba(234, 236, 240, 1)" paddingX={1} width="max-content" borderRadius={4} overflow="auto">
        <Typography variant="body2" component="code">
          {posRefId}
        </Typography>
      </Box>
      <Typography className={classes.body}>POS ID:</Typography>
      <Box bgcolor="rgba(234, 236, 240, 1)" paddingX={1} width="max-content" borderRadius={4} overflow="auto">
        <Typography variant="body2" component="code">
          {posId}
        </Typography>
      </Box>
      <Typography className={classes.body}>TID:</Typography>
      <Box bgcolor="rgba(234, 236, 240, 1)" paddingX={1} width="max-content" borderRadius={4} overflow="auto">
        <Typography variant="body2" component="code">
          {tid}
        </Typography>
      </Box>
      <Typography className={classes.body}>MID:</Typography>
      <Box bgcolor="rgba(234, 236, 240, 1)" paddingX={1} width="max-content" borderRadius={4} overflow="auto">
        <Typography variant="body2" component="code">
          {mid}
        </Typography>
      </Box>
    </div>
  );
};
