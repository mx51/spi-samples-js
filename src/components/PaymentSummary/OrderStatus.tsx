import { Divider, Typography } from '@material-ui/core';
import React from 'react';
import { SuccessState } from '@mx51/spi-client-js';
import dayjs from 'dayjs';
import { TxLogItem } from '../../services/txLogService';
import { useTransactionDetailPageStyle } from '../TransactionPage/TransactionDetailsPage/TransactionDetailsPage.style';
import { useGetTxDetails } from '../TransactionPage/useGetTxDetails';

type Props = {
  currentTransaction: TxLogItem;
};

export const OrderStatus = ({ currentTransaction }: Props) => {
  const classes = useTransactionDetailPageStyle();
  const { override, successState, type, completedTime } = currentTransaction;
  const { getIconByStatus, getStatus } = useGetTxDetails();

  return (
    <div>
      <div className={classes.iconContainer}>{getIconByStatus(override ? SuccessState.Unknown : successState)}</div>
      <Typography variant="h6" component="h1">
        {getStatus(override ? SuccessState.Unknown : successState)}
      </Typography>
      <Typography variant="h6" component="h1">
        {type.toUpperCase()}
      </Typography>
      <Typography className={classes.heading}>{dayjs(completedTime).format('hh:mm A')}</Typography>
      <Divider className={classes.divider} variant="middle" />
    </div>
  );
};
