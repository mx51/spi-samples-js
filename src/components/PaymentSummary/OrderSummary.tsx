import { Divider, Typography } from '@material-ui/core';
import React from 'react';
import OrderLineItem from '../OrderLineItem';
import OrderSubTotal from '../OrderSubTotal';
import currencyFormat from '../../utils/common/intl/currencyFormatter';
import { TxLogItem } from '../../services/txLogService';
import { useTransactionDetailPageStyle } from '../TransactionPage/TransactionDetailsPage/TransactionDetailsPage.style';

type Props = {
  currentTransaction: TxLogItem;
  isCashoutOnly: boolean;
};

export const OrderSummary = ({ currentTransaction, isCashoutOnly }: Props) => {
  const classes = useTransactionDetailPageStyle();
  const { amountCents, surchargeAmount, bankCashAmount, tipAmount, total } = currentTransaction;

  return (
    <div className={classes.summary}>
      <Divider className={classes.divider} variant="middle" />
      <Typography className={classes.heading}>Order Summary</Typography>
      <Typography variant="h6" component="h1" data-testid="total">
        {currencyFormat(total / 100)}
      </Typography>
      <OrderSubTotal label="Subtotal" amount={amountCents} data-testid="subTotal" />
      {surchargeAmount ? (
        <OrderLineItem disabled label="Surcharge" amount={surchargeAmount} viewOnly data-testid="surcharge" />
      ) : null}
      {!isCashoutOnly && bankCashAmount ? (
        <OrderLineItem
          disabled
          label="Cashout"
          amount={isCashoutOnly ? 0 : bankCashAmount}
          viewOnly
          data-testid="cashout"
        />
      ) : null}
      {tipAmount ? <OrderLineItem disabled label="Tip" amount={tipAmount} viewOnly data-testid="tip" /> : null}
    </div>
  );
};
