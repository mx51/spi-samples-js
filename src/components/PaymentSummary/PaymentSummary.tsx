import React, { useEffect, useMemo } from 'react';
import { Box, Button, Divider, Grid, Paper, Typography } from '@material-ui/core';
import { useDispatch } from 'react-redux';
import { Link as LinkRouter } from 'react-router-dom';
import useStyles from './index.style';
import {
  PATH_ACCOUNT_VERIFY,
  PATH_PRE_AUTH,
  TEXT_PRE_AUTH,
  PATH_PURCHASE,
} from '../../definitions/constants/routerConfigs';
import { ReactComponent as FailedIcon } from '../../images/FailedIcon.svg';
import { ReactComponent as SuccessIcon } from '../../images/SuccessIcon.svg';
import { clearAllProducts } from '../../redux/reducers/ProductSlice/productSlice';
import { ITxFlow } from '../../redux/reducers/TerminalSlice/interfaces';
import currencyFormat from '../../utils/common/intl/currencyFormatter';
import OrderLineItem from '../OrderLineItem';
import OrderSubTotal from '../OrderSubTotal';
import { getTxTypeByPosRefId } from '../../utils/tx-utils';

interface Props {
  typePath: string;
  isTxFinished: boolean;
  isTxSuccess: boolean;

  // TODO: confirm with product about the expected behaviour if the page is refresh
  // i.e no information on the last txFlow, posId, ...
  posId?: string;
  deviceAddress?: string;
  serialNumber?: string;
  txFlow: ITxFlow | null;
}

export const PaymentSummary: React.FC<Props> = ({
  typePath,
  isTxFinished,
  isTxSuccess,
  posId,
  deviceAddress,
  serialNumber,
  txFlow,
}) => {
  const classes = useStyles();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(clearAllProducts());
  }, []);

  const amountSummaryInformation = (type: string) => {
    const responseData = txFlow?.override ? txFlow?.request.data : txFlow?.response.data;
    if (responseData) return (responseData as Any)[type] ?? 0;
    return 0;
  };

  const pathNameFormatter = (path: string) => {
    if (path) {
      const pathName = path.split('/')[1];
      return pathName.charAt(0).toUpperCase() + pathName.slice(1);
    }
    return '';
  };

  const subTotal = amountSummaryInformation('purchaseAmount');
  const surchangeAmount = amountSummaryInformation('surchargeAmount');
  const cashoutAmount = amountSummaryInformation('bankCashAmount');
  const tipAmount = amountSummaryInformation('tipAmount');
  const refundAmount = amountSummaryInformation('refundAmount');

  const originalTotalAmount = [PATH_PRE_AUTH, PATH_ACCOUNT_VERIFY].includes(typePath)
    ? txFlow?.amountCents + surchangeAmount
    : subTotal + surchangeAmount + cashoutAmount + tipAmount + refundAmount;

  const transactionStatus = useMemo(() => {
    const transactionType = getTxTypeByPosRefId(txFlow?.posRefId ?? '').toUpperCase();
    const status = txFlow?.success === 'Success' ? 'APPROVED' : 'DECLINED';
    return `${transactionType} ${status}`;
  }, [txFlow]);

  return (
    <Box className={`${classes.root} ${typePath !== PATH_PURCHASE && classes.alignTop}`}>
      <Box flexGrow="2" className={classes.roots}>
        {isTxFinished && isTxSuccess && (
          <>
            <SuccessIcon data-testid="success-icon" className={classes.successIcon} />
            <Typography variant="h5" component="h1">
              {transactionStatus}
            </Typography>
          </>
        )}
        {isTxFinished && !isTxSuccess && (
          <>
            <FailedIcon data-testid="fail-icon" className={classes.failedIcon} />
            <Typography variant="h5" component="h1">
              {transactionStatus}
            </Typography>
          </>
        )}
        <Typography className={classes.heading}>Terminal: {posId}</Typography>

        <Typography className={classes.subheading}>
          {deviceAddress} | S/N {serialNumber}
        </Typography>
        <Box data-testid="total" className={classes.paper} component={Paper}>
          {currencyFormat((Number.isNaN(originalTotalAmount) ? 0 : originalTotalAmount) / 100)}
        </Box>
        {typePath === PATH_PURCHASE ? (
          <>
            <Typography className={classes.orderSummery}>Order Summary</Typography>
            <Divider variant="middle" />
            <OrderSubTotal
              data-testid="subTotal"
              label="Subtotal"
              amount={amountSummaryInformation('purchaseAmount')}
            />
            <OrderLineItem
              data-testid="surchage"
              disabled
              label="Surcharge"
              amount={amountSummaryInformation('surchargeAmount')}
              viewOnly
            />
            <OrderLineItem
              data-testid="cashout"
              disabled
              label="Cashout"
              amount={amountSummaryInformation('bankCashAmount')}
              viewOnly
            />
            <OrderLineItem
              data-testid="tip"
              disabled
              label="Tip"
              amount={amountSummaryInformation('tipAmount')}
              viewOnly
            />
          </>
        ) : (
          <br />
        )}
        <Grid container spacing={1}>
          {typePath !== PATH_PURCHASE && (
            <Grid item xs={6}>
              <Button
                variant="contained"
                color="primary"
                size="large"
                classes={{ root: classes.actionBtn }}
                component={LinkRouter}
                to={typePath !== PATH_ACCOUNT_VERIFY ? typePath : PATH_PRE_AUTH}
              >
                {typePath !== PATH_ACCOUNT_VERIFY ? pathNameFormatter(typePath) : TEXT_PRE_AUTH}
              </Button>
            </Grid>
          )}
          <Grid item xs={typePath !== PATH_PURCHASE ? 6 : 12}>
            <Button
              data-test-id="newOrderBtn"
              variant="contained"
              color="primary"
              size="large"
              classes={{ root: classes.actionBtn }}
              component={LinkRouter}
              to={PATH_PURCHASE}
            >
              New Order
            </Button>
          </Grid>
        </Grid>
      </Box>
    </Box>
  );
};
