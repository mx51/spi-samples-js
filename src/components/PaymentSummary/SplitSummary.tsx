import React from 'react';
import { Box, Paper } from '@material-ui/core';
import { useStyles } from '../SplitPage/SplitReceiptPanel/SplitReceiptPanel.styles';
import currencyFormat from '../../utils/common/intl/currencyFormatter';

type Props = {
  currentSplitNumber: number;
  totalSplitNumber: number;
  amount: number;
  outstandingAmount: number;
};

export const SplitSummary = ({ currentSplitNumber, totalSplitNumber, amount, outstandingAmount }: Props) => {
  const classes = useStyles();

  return (
    <Box className={classes.paper} component={Paper}>
      <span className={classes.splitNumberRow}>
        Split {currentSplitNumber + 1} of {totalSplitNumber}
      </span>
      {currencyFormat(amount / 100)}
      <Box className={classes.outstandingAmountRow}>
        <span>Outstanding amount</span>
        <span>{currencyFormat(outstandingAmount / 100)}</span>
      </Box>
    </Box>
  );
};
