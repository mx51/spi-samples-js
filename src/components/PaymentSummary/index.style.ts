import { alpha, createStyles, makeStyles, Theme } from '@material-ui/core';
import { FONT_COLOR } from '../../definitions/constants/themeStylesConfigs';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      textAlign: 'center',
      display: 'flex',
      alignItems: 'center',
      height: `calc(100vh - 60px)`,
      paddingLeft: theme.spacing(2),
      paddingRight: theme.spacing(2),
    },
    alignTop: {
      alignItems: 'unset',
      padding: theme.spacing(6, 2),
    },
    roots: {
      display: 'flex',
      flexDirection: 'column',
    },
    connectedIcon: {
      color: theme.palette.success.main,
      margin: theme.spacing(1),
      marginLeft: 'auto',
      marginRight: 'auto',
    },
    unpairedIcon: {
      color: theme.palette.error.main,
      margin: theme.spacing(1),
      marginLeft: 'auto',
      marginRight: 'auto',
    },

    heading: {
      fontSize: '1rem',
      fontWeight: 400,
      lineHeight: '1.5rem',
    },
    subheading: {
      fontSize: '.8rem',
      fontWeight: 400,
      lineHeight: '1.5rem',
      color: FONT_COLOR,
      marginBottom: theme.spacing(3),
    },
    paper: {
      border: 'none',
      alignItems: 'center',
      textAlign: 'center',
      padding: '1rem',
      letterSpacing: '0.02rem',
      fontSize: '3rem',
      fontWeight: 'normal',
    },
    orderSummery: {
      margin: theme.spacing(2),
      textAlign: 'left',
      paddingLeft: '.5rem',
      color: FONT_COLOR,
    },
    orderTotalInputField: {
      marginRight: theme.spacing(2),
      padding: '2rem',
      letterSpacing: '0.02rem',
      fontSize: '3rem',
      fontWeight: 'normal',
    },

    actionBtn: {
      padding: theme.spacing(2),
      borderRadius: 0,
      backgroundColor: theme.palette.error.dark,
      '&:hover, &:active, &:focus': {
        background: alpha(theme.palette.error.dark, 0.75),
        color: theme.palette.common.white,
      },
      width: '100%',
    },
  })
);
export default useStyles;
