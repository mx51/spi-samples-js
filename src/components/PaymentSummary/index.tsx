import React from 'react';
import { Box, Button, Divider, Grid, Paper, Typography } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import { Link as LinkRouter } from 'react-router-dom';
import { PATH_PURCHASE } from '../../definitions/constants/routerConfigs';
import { ReactComponent as FailedIcon } from '../../images/FailedIcon.svg';
import { ReactComponent as SuccessIcon } from '../../images/SuccessIcon.svg';

import { productSubTotalSelector } from '../../redux/reducers/ProductSlice/productSelector';
import { clearAllProducts } from '../../redux/reducers/ProductSlice/productSlice';
import selectedTerminalIdSelector from '../../redux/reducers/SelectedTerminalSlice/selectedTerminalSliceSelector';
import { ITerminalProps } from '../../redux/reducers/TerminalSlice/interfaces';
import {
  isTerminalTxFlowSuccess,
  terminalInstance,
  terminalTransactionTypeObject,
  terminalTxFlowFinishedTracker,
} from '../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import currencyFormat from '../../utils/common/intl/currencyFormatter';
import OrderLineItem from '../OrderLineItem';
import OrderSubTotal from '../OrderSubTotal';
import useStyles from './index.style';

function PaymentSummary(): React.ReactElement {
  const classes = useStyles();
  const dispatch = useDispatch();
  const selectedTerminal = useSelector(selectedTerminalIdSelector);
  const currentTerminal = useSelector(terminalInstance(selectedTerminal)) as ITerminalProps;
  const subtotalAmount = useSelector(productSubTotalSelector);
  const isTxFlowFinished = useSelector(terminalTxFlowFinishedTracker(selectedTerminal));
  const isTxFlowSuccess = useSelector(isTerminalTxFlowSuccess(selectedTerminal));
  const { typePath, typeTitle } = useSelector(terminalTransactionTypeObject(selectedTerminal));

  const clearAllProductsAction = () => {
    dispatch(clearAllProducts());
  };

  return (
    <Box className={classes.root}>
      <Box flexGrow="2" className={classes.roots}>
        {isTxFlowFinished && isTxFlowSuccess && (
          <>
            <SuccessIcon className={classes.successIcon} />
            <Typography variant="h5" component="h1">
              {typeTitle} Approved
            </Typography>
          </>
        )}
        {isTxFlowFinished && !isTxFlowSuccess && (
          <>
            <FailedIcon className={classes.failedIcon} />
            <Typography variant="h5" component="h1">
              {typeTitle} Declined
            </Typography>
          </>
        )}
        <Typography className={classes.heading}>Terminal:{currentTerminal?.posId}</Typography>

        <Typography className={classes.subheading}>
          {currentTerminal?.deviceAddress} S/N {currentTerminal?.serialNumber}
        </Typography>
        <Box className={classes.paper} component={Paper}>
          {currencyFormat((currentTerminal?.txFlow?.request.data.purchaseAmount ?? 0) / 100)}
        </Box>
        <Typography className={classes.orderSummery}>Order Summary</Typography>
        <Divider variant="middle" />
        <OrderSubTotal label="Subtotal" amount={subtotalAmount} />
        <OrderLineItem
          disabled
          label="Surcharge"
          onAdd={() => 0}
          amount={currentTerminal?.txFlow?.request.data.surchargeAmount ?? 0}
          viewOnly
        />
        <OrderLineItem
          disabled
          label="Cashout"
          onAdd={() => 0}
          amount={currentTerminal?.txFlow?.request.data.cashAmount ?? 0}
          viewOnly
        />
        <OrderLineItem
          disabled
          label="Tip"
          onAdd={() => 0}
          amount={currentTerminal?.txFlow?.request.data.tipAmount ?? 0}
          viewOnly
        />
        <Grid container spacing={1}>
          {typePath !== PATH_PURCHASE && (
            <Grid item xs={6}>
              <Button
                variant="contained"
                color="primary"
                size="large"
                classes={{ root: classes.actionBtn }}
                component={LinkRouter}
                to={typePath}
              >
                {typeTitle}
              </Button>
            </Grid>
          )}
          <Grid item xs={typePath !== PATH_PURCHASE ? 6 : 12}>
            <Button
              data-test-id="newOrderBtn"
              variant="contained"
              color="primary"
              size="large"
              classes={{ root: classes.actionBtn }}
              component={LinkRouter}
              to={PATH_PURCHASE}
              onClick={clearAllProductsAction}
            >
              New Order
            </Button>
          </Grid>
        </Grid>
      </Box>
    </Box>
  );
}

export default PaymentSummary;
