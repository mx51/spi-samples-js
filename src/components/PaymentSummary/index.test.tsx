import React from 'react';
import { cleanup, fireEvent } from '@testing-library/react';
import PaymentSummary from '.';
import { TxFlowState } from '../../definitions/constants/terminalConfigs';
import { clearAllProducts } from '../../redux/reducers/ProductSlice/productSlice';
import { store } from '../../redux/store';
import mockWithRedux, {
  defaultMockPairFormParams,
  mockReceiptResponse,
  mockRefundTxFlow,
  mockTerminalInstanceId,
  pairedMockTerminals,
} from '../../utils/tests/common';

describe('Test <PaymentSummary />', () => {
  let mockContainer: Any;
  let customStore: Any;

  beforeEach(() => {
    mockContainer = mockWithRedux(<PaymentSummary />);
    customStore = (mockTerminals: Any) => ({
      ...store,
      getState: () => ({
        common: { showFlowPanel: false, acquireConfirmPairingFlow: false },
        pairForm: defaultMockPairFormParams,
        terminals: mockTerminals,
        products: {
          surchargeAmount: 100,
          tipAmount: 100,
          cashoutAmount: 100,
          products: [],
        },
        selectedTerminal: { selectedTerminalId: mockTerminalInstanceId },
      }),
    });
  });

  afterEach(cleanup);

  test('should match PaymentSummary snapshot test', () => {
    // Assert
    expect(mockContainer).toMatchSnapshot();
  });

  test('should call clearAllProducts() when new order button get clicked', () => {
    // Arrange
    const newOrderBtnDOM = mockContainer.querySelector('[data-test-id="newOrderBtn"]');
    const dispatch = jest.fn();
    dispatch(clearAllProducts());
    const actionBtnClassOccurrence = (mockContainer.innerHTML.match(/actionBtn/g) || []).length;

    // Act
    fireEvent.click(newOrderBtnDOM);

    // Assert
    expect(dispatch).toHaveBeenCalled();
    expect(actionBtnClassOccurrence).toBe(1);
  });

  test('should show approved text when transaction has been done successfully', () => {
    // Arrange
    const mockTerminals = {
      [mockTerminalInstanceId]: {
        ...pairedMockTerminals[mockTerminalInstanceId],
        receipt: mockReceiptResponse,
        txFlow: mockRefundTxFlow,
      },
    };
    const refundApprovedText = 'Refund Approved';
    const newMockContainer = mockWithRedux(<PaymentSummary />, customStore(mockTerminals));
    const actionBtnClassOccurrence = (newMockContainer.innerHTML.match(/actionBtn/g) || []).length;

    // Assert
    expect(newMockContainer.innerHTML.includes(refundApprovedText)).toBeTruthy();
    expect(actionBtnClassOccurrence).toBe(2);
  });

  test('should show declined text when transaction has failed', () => {
    // Arrange
    const mockTerminals = {
      [mockTerminalInstanceId]: {
        ...pairedMockTerminals[mockTerminalInstanceId],
        receipt: mockReceiptResponse,
        txFlow: {
          ...mockRefundTxFlow,
          success: TxFlowState.Failed,
        },
      },
    };

    const refundDeclinedText = 'Refund Declined';
    const newMockContainer = mockWithRedux(<PaymentSummary />, customStore(mockTerminals));
    const actionBtnClassOccurrence = (newMockContainer.innerHTML.match(/actionBtn/g) || []).length;

    // Assert
    expect(newMockContainer.innerHTML.includes(refundDeclinedText)).toBeTruthy();
    expect(actionBtnClassOccurrence).toBe(2);
  });
});
