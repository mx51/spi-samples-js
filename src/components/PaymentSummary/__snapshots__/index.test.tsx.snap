// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PaymentSummary /> should match PaymentSummary snapshot test 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-12 makeStyles-root-1 false"
  >
    <div
      class="MuiBox-root MuiBox-root-13 makeStyles-roots-3"
    >
      <p
        class="MuiTypography-root makeStyles-heading-6 MuiTypography-body1"
      >
        Terminal:
      </p>
      <p
        class="MuiTypography-root makeStyles-subheading-7 MuiTypography-body1"
      >
         S/N 
      </p>
      <div
        class="MuiPaper-root MuiBox-root MuiBox-root-14 makeStyles-paper-8 MuiPaper-elevation1 MuiPaper-rounded"
      >
        $0.00
      </div>
      <p
        class="MuiTypography-root makeStyles-orderSummery-9 MuiTypography-body1"
      >
        Order Summary
      </p>
      <hr
        class="MuiDivider-root MuiDivider-middle"
      />
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-subTotal-15 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Subtotal
          </span>
        </div>
        <p
          class="MuiTypography-root makeStyles-price-16 MuiTypography-body1"
        >
          $0.00
        </p>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesDisable-18 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Surcharge
          </span>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          $0.00
        </p>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesDisable-18 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Cashout
          </span>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          $0.00
        </p>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesDisable-18 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Tip
          </span>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          $0.00
        </p>
      </li>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root makeStyles-actionBtn-11 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            data-test-id="newOrderBtn"
            href="/"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              New Order
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test <PaymentSummary /> should match snapshot for Purchase failure 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-136 makeStyles-root-115 makeStyles-alignTop-116"
  >
    <div
      class="MuiBox-root MuiBox-root-137 makeStyles-roots-117"
    >
      <svg
        class="makeStyles-failedIcon-119"
      >
        FailedIcon.svg
      </svg>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Refund
         Declined
      </h1>
      <p
        class="MuiTypography-root makeStyles-heading-120 MuiTypography-body1"
      >
        Terminal:
        123
      </p>
      <p
        class="MuiTypography-root makeStyles-subheading-121 MuiTypography-body1"
      >
        192.168.1.10
         S/N 
        123-123-123
      </p>
      <div
        class="MuiPaper-root MuiBox-root MuiBox-root-138 makeStyles-paper-122 MuiPaper-elevation1 MuiPaper-rounded"
      >
        $5.50
      </div>
      <br />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root makeStyles-actionBtn-125 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            href="/refund"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Refund
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root makeStyles-actionBtn-125 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            data-test-id="newOrderBtn"
            href="/"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              New Order
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test <PaymentSummary /> should match snapshot for Purchase success 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-112 makeStyles-root-91 makeStyles-alignTop-92"
  >
    <div
      class="MuiBox-root MuiBox-root-113 makeStyles-roots-93"
    >
      <svg
        class="makeStyles-successIcon-94"
      >
        SuccessIcon.svg
      </svg>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Refund
         Approved
      </h1>
      <p
        class="MuiTypography-root makeStyles-heading-96 MuiTypography-body1"
      >
        Terminal:
        123
      </p>
      <p
        class="MuiTypography-root makeStyles-subheading-97 MuiTypography-body1"
      >
        192.168.1.10
         S/N 
        123-123-123
      </p>
      <div
        class="MuiPaper-root MuiBox-root MuiBox-root-114 makeStyles-paper-98 MuiPaper-elevation1 MuiPaper-rounded"
      >
        $5.50
      </div>
      <br />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root makeStyles-actionBtn-101 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            href="/refund"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Refund
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root makeStyles-actionBtn-101 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            data-test-id="newOrderBtn"
            href="/"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              New Order
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
